From 026621a7dfa8e059d1767fa1bfcc340dd14afa66 Mon Sep 17 00:00:00 2001
From: Lukasz Majewski <lukma@denx.de>
Date: Wed, 10 Mar 2021 11:54:26 +0100
Subject: [PATCH 3/5] dts: Re-enable reset for mv88e6020 switch

As the PHY's are powered up in U-Boot (as the NO_CPU bootstrap is 0),
it is welcome to reset the switch in Linux to allow full initialization
by the mv88e6xxx driver.

The GPIO configuration is not touched in Linux driver - we keep
the PAD configuration from U-Boot.

Signed-off-by: Lukasz Majewski <lukma@denx.de>
---
 arch/arm/boot/dts/imx6ull-kie-inverter.dts | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/imx6ull-kie-inverter.dts b/arch/arm/boot/dts/imx6ull-kie-inverter.dts
index 60e9e7e6b764..49266c6b93b8 100644
--- a/arch/arm/boot/dts/imx6ull-kie-inverter.dts
+++ b/arch/arm/boot/dts/imx6ull-kie-inverter.dts
@@ -150,10 +150,9 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_enet2>;
 	phy-mode = "rmii";
-	phy-handle = <&ethphy1>;
-	//phy-reset-gpios = <&gpio3 22 GPIO_ACTIVE_LOW>;
-	//phy-reset-duration = <1>;
-	//phy-reset-post-delay = <1>;
+	phy-reset-gpios = <&gpio3 22 GPIO_ACTIVE_LOW>;
+	phy-reset-duration = <10>;
+	phy-reset-post-delay = <20>;
 	phy-supply = <&reg_3v3>;
 	status = "okay";
 
@@ -271,6 +270,7 @@
 			MX6UL_PAD_ENET2_TX_DATA0__ENET2_TDATA00	0x1b010
 			MX6UL_PAD_ENET2_TX_DATA1__ENET2_TDATA01	0x1b010
 			MX6UL_PAD_ENET2_TX_CLK__ENET2_REF_CLK2	0x4001b031
+			MX6UL_PAD_LCD_DATA17__GPIO3_IO22        0x80000000
 		>;
 	};
 
-- 
2.20.1

